---
title: "Spatial vs. temporal storage: The dual role of microbial seed
banks in driving geographical patterns of microbial diversity"

geometry: margin=2.54cm
date: "August 4, 2015"
output: pdf_document
header-includes: \usepackage{array}
---

```{r, results = 'hide', echo=FALSE, message = FALSE, warning = FALSE}
# Retrieve and set the working directory
rm(list=ls())
getwd()
setwd("~/GitHub/Dimensions/Aim3")
```


```{r, results = 'hide', echo=FALSE, message = FALSE, warning = FALSE}
# Load packages
require("picante")
require("ape")
require("seqinr")
require("vegan") # biodiversity estimators and related functions
require("fossil")
require("simba")
require("reshape")
```


```{r, results = 'hide', echo=FALSE, message = FALSE, warning = FALSE}
# Import functions
# import code file that includes a function for reading in output files 
#from community sequencing software `mothur` (http://www.mothur.org/).
source("~/GitHub/Dimensions/Aim3/bin/MothurTools.R")
load(file = "~/GitHub/Dimensions/Aim3/Mothur/INPond_Initial.RData")
```


```{r, results = 'hide', echo=FALSE, message = FALSE, warning = FALSE}
# Load Environmental and Geographical Data
env <- read.table("~/GitHub/Dimensions/Aim3/DATA/EnvData/20130801_PondDataMod.csv",
                  sep = ",", header = TRUE)

# Remove 189, 190, 191
# 236, 237 missing lots of data (very big ponds, outliers of sorts)
env <- env[complete.cases(env),]

# Select DNA Data: Use the `grep()` Command and Rename with `gsub()`
# The active portion, based on cDNA 
active.comm <- Pond97[grep("*-cDNA", rownames(Pond97)), ]
rownames(active.comm) <- gsub("\\-cDNA", "", rownames(active.comm))
rownames(active.comm) <- gsub("\\_", "", rownames(active.comm))

# The community without respect to active or not, 16S rRNA gene sequences
all.comm <- Pond97[grep("*-DNA", rownames(Pond97)), ]
rownames(all.comm) <- gsub("\\-DNA", "", rownames(all.comm))
rownames(all.comm) <- gsub("\\_", "", rownames(all.comm))

# Remove Sites Not in the Environmental Data Set
active.comm <- active.comm[rownames(active.comm)  %in% env$Sample_ID, ]
all.comm <- all.comm[rownames(all.comm)  %in% env$Sample_ID, ]

# Remove sites not shared between matrices
df <- as.data.frame(active.comm)
all.comm <- all.comm[rownames(all.comm)  %in% row.names(df),]
env <- env[env$Sample_ID %in% row.names(df),]

# Geographic Distances (Kilometers) Among Ponds
long.lat <- as.matrix(cbind(env$long, env$lat))
geo.dist <- earth.dist(long.lat, dist = TRUE)
geo.dist[which(!is.finite(geo.dist))] = NA

# Remove Zero-Occurrence Taxa 
active.comm <- active.comm[ , colSums(active.comm) > 0]
all.comm <- all.comm[ , colSums(all.comm) > 0]

active.comm <- active.comm + 1
active.comm <- log10(active.comm)
all.comm <- all.comm + 1
all.comm <- log10(all.comm)

cutoff <- 4000
all.comm <- all.comm[, 1:cutoff]
active.comm <- active.comm[, 1:cutoff]

# Geographic variables
geo.dat <- as.matrix(subset(env, select = lat:long))
# Pond environmental variables
env.dat <- as.matrix(subset(env, select = Depth:TP))
```


```{r}
metric <- "jaccard"
bin <- F

levels.short <- c('BCSP', 'HNF', 'YSF')
colors.short <- c('firebrick1', 'darkorchid1', 'cornflowerblue')
  
colors <- c('firebrick1','firebrick1','firebrick1','firebrick1','firebrick1',
            'firebrick1','firebrick1','firebrick1','firebrick1','firebrick1',
            'firebrick1','firebrick1','firebrick1','firebrick1','firebrick1',
            'firebrick1', 'firebrick1',
            'darkorchid1','darkorchid1','darkorchid1','darkorchid1','darkorchid1',
            'darkorchid1','darkorchid1','darkorchid1','darkorchid1','darkorchid1',
            'darkorchid1','darkorchid1','darkorchid1','darkorchid1','darkorchid1',
            'darkorchid1', 'darkorchid1',
            'cornflowerblue','cornflowerblue','cornflowerblue','cornflowerblue',
            'cornflowerblue', 'cornflowerblue','cornflowerblue','cornflowerblue',
            'cornflowerblue','cornflowerblue', 'cornflowerblue','cornflowerblue',
            'cornflowerblue','cornflowerblue','cornflowerblue',
            'firebrick1', 'darkorchid1', 'cornflowerblue','cornflowerblue',
            'darkorchid1','darkorchid1','darkorchid1','darkorchid1','darkorchid1')

# Conduct CCA 
active.cca <- vegan::cca(active.comm ~ env.dat)
all.cca <- vegan::cca(all.comm ~ env.dat)
#active.cca <- vegan::cca(env.dat ~ active.comm)
#all.cca <- vegan::cca(env.dat ~ all.comm)
```



```{r}
file <- paste("~/GitHub/Dimensions/Aim3/papers/DD/figs/CCA-active.png", sep="")
png(filename=file)

cca.explainvar1 <- round(active.cca$CCA$eig[1] / 
                         sum(c(active.cca$CCA$eig, active.cca$CA$eig)), 3) * 100
cca.explainvar2 <- round(active.cca$CCA$eig[2] / 
                         sum(c(active.cca$CCA$eig, active.cca$CA$eig)), 3) * 100

# Define Plot Parameters
par(mfrow=c(1, 1), mar = c(5, 5, 4, 4) + 0.1, pty="s")

# Initiate Plot
plot(scores(active.cca, display = "wa"), xlim = c(-2.7, 2.7), ylim = c(-2.5, 2.5),
     xlab = paste("CCA 1 (", cca.explainvar1, "%)", sep = ""),
     ylab = paste("CCA 2 (", cca.explainvar2, "%)", sep = ""),
     pch = 16, cex = 2.0, type = "n", cex.lab = 1.5, cex.axis = 1.2,
     axes = FALSE)

# Add Axes
axis(side = 1, labels = T, lwd.ticks = 2, cex.axis = 1.2, las = 1)
axis(side = 2, labels = T, lwd.ticks = 2, cex.axis = 1.2, las = 1)
abline(h = 0, v = 0, lty = 3)
box(lwd = 2)

# Add Points & Labels
points(scores(active.cca, display = "wa"),
       pch = 1, cex = 2, bg = colors, col = colors)
legend("topleft", legend = levels.short, bty = "n", col = colors.short, 
       pch = 1)

# Add Environmental Vectors
vectors <- scores(active.cca, display = "bp")
row.names(vectors) <- colnames(env.dat)

arrows(0, 0, vectors[,1] * 3, vectors[, 2] * 3, 
       lwd = 1, lty = 1, length = 0.2, col = "black")

text(vectors[,1] * 3.5, vectors[, 2] * 3.5, pos = 3, 
     labels = row.names(vectors))

axis(side = 3, lwd.ticks=2, cex.axis=1.2, las = 1, col = "red", lwd = 2.2,
     at = pretty(range(vectors[, 1])) * 2, labels = pretty(range(vectors[, 1])))
axis(side = 4, lwd.ticks=2, cex.axis=1.2, las = 1, col = "red", lwd = 2.2,
     at = pretty(range(vectors[, 2])) * 2, labels = pretty(range(vectors[, 2])))

dev.off()
```



```{r}
file <- paste("~/GitHub/Dimensions/Aim3/papers/DD/figs/CCA-all.png", sep="")
png(filename=file)

# Permutation Tests
#anova(active.cca, by = "axis")
#cca.fit <- envfit(active.cca, env.dat, perm = 999)

# Calculate Explained Variation
cca.explainvar1 <- round(all.cca$CCA$eig[1] / 
                         sum(c(all.cca$CCA$eig, all.cca$CA$eig)), 3) * 100
cca.explainvar2 <- round(all.cca$CCA$eig[2] / 
                         sum(c(all.cca$CCA$eig, all.cca$CA$eig)), 3) * 100

# Define Plot Parameters
par(mfrow=c(1, 1), mar = c(5, 5, 4, 4) + 0.1, pty="s")

# Initiate Plot
plot(scores(all.cca, display = "wa"), xlim = c(-2.5, 3), ylim = c(-2.0, 2.0),
     xlab = paste("CCA 1 (", cca.explainvar1, "%)", sep = ""),
     ylab = paste("CCA 2 (", cca.explainvar2, "%)", sep = ""),
     pch = 16, cex = 2.0, type = "n", cex.lab = 1.5, 
     cex.axis = 1.2, axes = FALSE)

# Add Axes
axis(side = 1, labels = T, lwd.ticks = 2, cex.axis = 1.2, las = 1)
axis(side = 2, labels = T, lwd.ticks = 2, cex.axis = 1.2, las = 1)
abline(h = 0, v = 0, lty = 3)
box(lwd = 2)

# Add Points & Labels
points(scores(all.cca, display = "wa"),
       pch = 1, cex = 2, bg = colors, col = colors)
legend("topleft", legend = levels.short, bty = "n", 
       col = colors.short, pch = 1)

# Add Environmental Vectors
vectors <- scores(all.cca, display = "bp")
row.names(vectors) <- colnames(env.dat)


arrows(0, 0, vectors[,1] * 3, vectors[, 2] * 3, 
       lwd = 1, lty = 1, length = 0.2, col = "black")

text(vectors[, 1] * 3.5, vectors[, 2] * 3.5, pos = 3, 
     labels = row.names(vectors))

axis(side = 3, lwd.ticks=2, cex.axis=1.2, las = 1, col = "red", lwd = 2.2,
     at = pretty(range(vectors[, 1])) * 2, labels = pretty(range(vectors[, 1])))
axis(side = 4, lwd.ticks=2, cex.axis=1.2, las = 1, col = "red", lwd = 2.2,
     at = pretty(range(vectors[, 2])) * 2, labels = pretty(range(vectors[, 2])))

dev.off()
```



```{r, results = "hide"}
active.cca.scores <- scores(active.cca)
active.cca.sites <- as.data.frame(scores(active.cca.scores$sites))
active.cca.scores <- as.matrix(subset(active.cca.sites, select = CCA1))
active.cca.dist <- vegdist(active.cca.scores, "euclidean")

all.cca.scores <- scores(all.cca)
all.cca.sites <- as.data.frame(scores(all.cca.scores$sites))
all.cca.scores <- as.matrix(subset(all.cca.sites, select = CCA1))
all.cca.dist <- vegdist(all.cca.scores, "euclidean")
```



```{r, results = "hide"}
x.lab <- "Difference in CCA1 score"

#file <- paste("~/GitHub/Dimensions/Aim3/papers/DD/figs/", 
#              metric,"_CCA1_Dist.png", sep="")
file <- paste("~/GitHub/Dimensions/Aim3/papers/DD/figs/sorensen_CCA1_Dist.png", sep="")

png(filename=file, units="in", width=8, height=8, res=600)
plot.new()
par(mfrow=c(1, 2), mar = c(5, 5, 2, 2) + 0.1, oma=c(0,0,2,0), pty="s")

active.dist <- 1 - vegdist(active.comm, method=metric, binary=bin) 
#active.dist <- log10(active.dist)
all.dist <- 1 - vegdist(all.comm, method=metric, binary=bin)
#all.dist <- log10(all.dist)

# Regression for active presence/absence
DD.active.presabs <- lm(active.dist ~ active.cca.dist)

# Regression for all presence/absence 
DD.all.presabs <- lm(all.dist ~ all.cca.dist)
  
# Make Plot for all presence/absence
slope <- round(coefficients(DD.all.presabs)[2], 3)
p <- round(summary(DD.all.presabs)$coefficients[8],3)
plot(all.cca.dist, all.dist, xaxt = "s", las = 1, 
   ylab="Community similarity", xlab=x.lab, col = "azure4",
   cex.lab=1, ylim = c(0.2, 0.7))
abline(DD.all.presabs , col = "red", lwd = 3)
legend("topleft", paste("All\nslope=",slope),
       bty="n", cex=1)

# Make Plot for active presence/absence
slope <- round(coefficients(DD.active.presabs)[2], 3)
p <- round(summary(DD.active.presabs)$coefficients[8],3)
plot(active.cca.dist, active.dist, xaxt = "s", las = 1, 
   ylab="Community similarity", xlab=x.lab, col = "azure4",
   cex.lab=1, ylim = c(0.2, 0.7))
abline(DD.active.presabs , col = "red", lwd = 3)
legend("topleft", paste("Active\nslope = ", slope),
       bty="n", cex=1)

# Add X-Axis Label to Plot
mtext("Geographic Distance, km", side = 1, adj = 0, outer = TRUE)
  
#d <- diffslope(active.cca.dist, active.dist, all.cca.dist, all.dist)
#Mtitle <- paste('Unweighted: Difference in slope =', 
#                round(as.numeric(d[2]),3), '; p =', d[3])
#print(Mtitle)
#title(main=Mtitle, line=-2, outer=T, cex.main=1.5)

dev.off()
```



```{r, results = "hide"}
metric <- "bray"
x.lab <- "Geographic distance, km"

file <- paste("~/GitHub/Dimensions/Aim3/papers/DD/figs/",
              metric,"_Geo_Dist.png", sep="")
  
png(filename=file, units="in", width=8, height=8, res=600)
plot.new()
par(mfrow=c(1, 2), mar = c(5, 5, 2, 2) + 0.1, oma=c(0,0,2,0), pty="s")


# Taxonomic Distances Among Ponds
active.dist <- 1 - vegdist(active.comm, method=metric, binary=bin) 
all.dist <- 1 - vegdist(all.comm, method=metric, binary=bin)

# Regression for active presence/absence
DD.active.presabs <- lm(active.dist ~ geo.dist)

# Regression for all presence/absence 
DD.all.presabs <- lm(all.dist ~ geo.dist)
  
# Make Plot for all presence/absence
slope <- round(coefficients(DD.all.presabs)[2], 3)
p <- round(summary(DD.all.presabs)$coefficients[8],3)
plot(geo.dist, all.dist, xaxt = "s", las = 1, 
   ylab="Community similarity", xlab=x.lab, col = "azure4",
   cex.lab=1, ylim=c(0.1, 0.7))
abline(DD.all.presabs , col = "red", lwd = 3)
legend("topleft", paste("All\nslope=",slope),
       bty="n", cex=1)

# Make Plot for active presence/absence
slope <- round(coefficients(DD.active.presabs)[2], 3)
p <- round(summary(DD.active.presabs)$coefficients[8],3)
plot(geo.dist, active.dist, xaxt = "s", las = 1, 
   ylab="Community similarity", xlab=x.lab, col = "azure4",
   cex.lab=1, ylim=c(0.1, 0.7))
abline(DD.active.presabs , col = "red", lwd = 3)
legend("topleft", paste("Active\nslope = ", 
                          slope), bty="n", cex=1)

# Add X-Axis Label to Plot
mtext("Geographic Distance, km", side = 1, adj = 0, outer = TRUE)
#d <- diffslope(active.cca.dist, active.dist, active.cca.dist, all.dist)

dev.off()
```