---
title: "Distance Decay Relationship among microbial communities of IN Ponds"
geometry: margin=2.54cm
date: "November 7, 2015"
output: pdf_document
header-includes: \usepackage{array}
---

```{r}
# Retrieve and set the working directory
rm(list=ls())
getwd()
setwd("~/GitHub/DistDecay/empirical_analyses/DistanceDecay")
# Load packages
require("simba")
require("fossil")
```


```{r}
# Load community data
load(file = "~/GitHub/DistDecay/data/MicrobialCommunity/INPond_Initial.RData")
# Load Environmental and Geographical Data
env <- read.table("~/GitHub/DistDecay/data/Environmental/20130801_INPondDataMod.csv", sep = ",", header = TRUE)
env <- env[complete.cases(env),]
env[which(env$Sample_ID == "HNF133"), ]["SpC"] <- 55
env[which(env$Sample_ID == "YSF46"), ]["lat"] <- 39.1186
```


```{r, results = 'hide', echo=FALSE, message = FALSE, warning = FALSE}
# Select DNA Data: Use the `grep()` Command and Rename with `gsub()`
# The active portion, based on cDNA
act.com <- Pond97[grep("*-cDNA", rownames(Pond97)), ]
rownames(act.com) <- gsub("\\-cDNA", "", rownames(act.com))

# The community without respect to active or not, 16S rRNA gene sequences
all.com <- Pond97[grep("*-DNA", rownames(Pond97)), ]
rownames(all.com) <- gsub("\\-DNA", "", rownames(all.com))

# Remove Sites Not in the Environmental Data Set
act.com <- act.com[rownames(act.com)  %in% env$Sample_ID, ]
all.com <- all.com[rownames(all.com)  %in% env$Sample_ID, ]

# Remove sites not shared between matrices
all.com <- all.com[rownames(all.com)  %in% row.names(act.com),]
env <- env[env$Sample_ID %in% row.names(act.com),]

# Remove Zero-Occurrence Taxa
act.com <- act.com[ , colSums(act.com) > 0]
all.com <- all.com[ , colSums(all.com) > 0]

# Use relative abundance
#act.com <- sweep(act.com, 1, rowSums(act.com), '/')
#all.com <- sweep(all.com, 1, rowSums(all.com), '/')
```

```{r}
rowMeans(act.com)
max(act.com)
max(all.com)
rowMeans(all.com)
mean(rowSums(act.com != 0))#/21568
mean(rowSums(all.com != 0))#/27466


```

```{r}
# Geographic Distances (Kilometers) Among Ponds
long.lat <- as.matrix(cbind(env$long, env$lat))
geo.dist <- earth.dist(long.lat, dist = TRUE)
geo.dist <- geo.dist/max(geo.dist)
geo.dist[which(!is.finite(geo.dist))] = NA

# Define Environmental Matrix
env.dat <- as.matrix(subset(env, select = c("canopy", "DO", "ORP", "pH", "Temp", "DOC", "TDS", "Color", "chla", "TP", "Depth", "Salinity", "SpC", "DON")))
env.dat <- scale(env.dat, center = TRUE, scale = TRUE)

# Conduct PCA
pca <- princomp(env.dat)
scores <- as.data.frame(pca$scores)
env.dist <- vegdist(scores[, 1], "euclidean")
env.dist <- env.dist/max(env.dist)
env.dist[which(!is.finite(env.dist))] = NA

write.table(scores[,1], "~/GitHub/DistDecay/model/EmpiricalDataForModeling/PCA-data/pca.txt", sep="\t")
write.table(env$long, "~/GitHub/DistDecay/model/EmpiricalDataForModeling/PCA-data/lon.txt", sep="\t")
write.table(env$lat, "~/GitHub/DistDecay/model/EmpiricalDataForModeling/PCA-data/lat.txt", sep="\t")
```


###################################################
# Compare Active & All
###################################################

```{r}
# Taxonomic Distances Among Ponds
act.bray.dist <- 1 - vegdist(act.com, method="bray", binary=F)
all.bray.dist <- 1 - vegdist(all.com, method="bray", binary=F)
act.sore.dist <- 1 - vegdist(act.com, method="bray", binary=T)
all.sore.dist <- 1 - vegdist(all.com, method="bray", binary=T)
act.canb.dist <- 1 - vegdist(act.com, method="canberra", binary=F)
all.canb.dist <- 1 - vegdist(all.com, method="canberra", binary=F)
```


```{r, results = 'hide', echo=FALSE, message = FALSE, warning = FALSE, fig.width=6, fig.height=6}

# Regression for active weighted, environmental distance
DD.act.env <- lm(act.bray.dist ~ env.dist)
#summary(DD.act.env, correlation = TRUE)

# Regression for all weighted, environmental distance
DD.all.env <- lm(all.bray.dist ~ env.dist)
coeff <- summary(DD.all.env, correlation = TRUE)

# Regression for active weighted, geographic distance
DD.act.geo <- lm(act.bray.dist ~ geo.dist)
#summary(DD.act.geo, correlation = TRUE)

# Regression for all weighted, geographic distance
DD.all.geo <- lm(all.bray.dist ~ geo.dist)
coeff <- summary(DD.all.geo, correlation = TRUE)

clr <- colorRampPalette(c("#00007F", "blue", "#007FFF", "cyan",
              "#7FFF7F", "yellow", "#FF7F00", "red", "#7F0000"))

file <- paste("~/GitHub/DistDecay/figs/FromSites/DDR-Bray.png", sep="")
png(filename=file, width=5, height=5, units="in", res=600, pointsize=10)
par(mfrow=c(2, 2), mar = c(5,5,3,1), oma =c(0,0,4,0))

# Make Plot for active pres-abs (env)
slope1 <- round(coefficients(DD.act.env)[2], 3)
int1 <- round(coefficients(DD.act.env)[1], 3)
p <- round(summary(DD.act.env)$coefficients[8],3)
colors <- densCols(env.dist, act.bray.dist, colramp=clr)
plot(env.dist, act.bray.dist, ylab="Similarity", pch=20,
  xlab="Environmental distance", col = colors, cex=1, cex.lab=1.2)
title(paste("Active: slope = ", slope1), line = 1, cex=1)
abline(DD.act.env, col = "black", lwd=3)


# Make Plot for all pres-abs (env)
slope2 <- round(coefficients(DD.all.env)[2], 3)
int2 <- round(coefficients(DD.all.env)[1], 3)
p <- round(summary(DD.all.env)$coefficients[8], 3)
colors <- densCols(env.dist, all.bray.dist, colramp=clr)
plot(env.dist, all.bray.dist, ylab="Similarity", pch=20,
  xlab="Environmental distance", col = colors, cex=1, cex.lab=1.2)
title(paste("All: slope = ", slope2), line = 1, cex=2)
abline(DD.all.env, col = "black", lwd=3)

# Make Plot for active pres-abs (geo)
slope3 <- round(coefficients(DD.act.geo)[2], 3)
int3 <- round(coefficients(DD.act.geo)[1], 3)
p <- round(summary(DD.act.geo)$coefficients[8],3)
colors <- densCols(geo.dist, act.bray.dist, colramp=clr)
plot(geo.dist, act.bray.dist, ylab="Similarity", pch=20,
  xlab="Geographic distance", col = colors, cex=1, cex.lab=1.2)
title(paste("Active: slope = ", slope3), line = 1, cex=1)
abline(DD.act.geo, col = "black", lwd=3)


# Make Plot for all pres-abs (geo)
slope4 <- round(coefficients(DD.all.geo)[2], 3)
int4 <- round(coefficients(DD.all.geo)[1], 3)
p <- round(summary(DD.all.geo)$coefficients[8], 3)
colors <- densCols(geo.dist, all.bray.dist, colramp=clr)
plot(geo.dist, all.bray.dist, ylab="Similarity", pch=20,
  xlab="Geographic distance", col = colors, cex=1, cex.lab=1.2)
title(paste("All: slope = ", slope4), line = 1, cex=1)
abline(DD.all.geo, col = "black", lwd=3)



# Add X-Axis Label to Plot
a.dd <- 'y'
if(a.dd == 'y'){
  mtext("Geographic Distance", side = 1, adj = 0, outer = TRUE)
  d1 <- diffslope(env.dist, act.bray.dist, env.dist, all.bray.dist, ic=TRUE, permutations=1000)
  d2 <- diffslope(geo.dist, act.bray.dist, geo.dist, all.bray.dist, ic=TRUE, permutations=1000)
  
  Mtitle <- paste("Bray-Curtis\n",
      round(100*(abs((slope1-slope2)/mean(c(slope1, slope2)))),1), '% difference in environmental DDRs; p = ', d1[3],
      '\n',round(100*(abs((slope3-slope4)/mean(c(slope3, slope4)))),1), '% difference in geographic DDRs; p = ', d2[3], sep="")
  title(main=Mtitle, line=0, outer=T, cex.main=1.2)
}
dev.off()

print(c('act.env', 'all.env', 'act.geo', 'all.geo'))
print(c(slope1, slope2, slope3, slope4))
print(c(int1, int2, int3, int4))

100*(abs((int1-int2)/mean(c(int1, int2))))
100*(abs((int3-int4)/mean(c(int3, int4))))
```


```{r, results = 'hide', echo=FALSE, message = FALSE, warning = FALSE, fig.width=6, fig.height=6}

# Regression for active weighted, environmental distance
DD.act.env <- lm(act.sore.dist ~ env.dist)
#summary(DD.act.env, correlation = TRUE)

# Regression for all weighted, environmental distance
DD.all.env <- lm(all.sore.dist ~ env.dist)
coeff <- summary(DD.all.env, correlation = TRUE)

# Regression for active weighted, geographic distance
DD.act.geo <- lm(act.sore.dist ~ geo.dist)
#summary(DD.act.geo, correlation = TRUE)

# Regression for all weighted, geographic distance
DD.all.geo <- lm(all.sore.dist ~ geo.dist)
coeff <- summary(DD.all.geo, correlation = TRUE)

clr <- colorRampPalette(c("#00007F", "blue", "#007FFF", "cyan",
              "#7FFF7F", "yellow", "#FF7F00", "red", "#7F0000"))

file <- paste("~/GitHub/DistDecay/figs/FromSites/DDR-Sorensen.png", sep="")
png(filename=file, width=5, height=5, units="in", res=600, pointsize=10)
par(mfrow=c(2, 2), mar = c(5,5,3,1), oma =c(0,0,4,0))

# Make Plot for active pres-abs (env)
slope1 <- round(coefficients(DD.act.env)[2], 3)
int1 <- round(coefficients(DD.act.env)[1], 3)
p <- round(summary(DD.act.env)$coefficients[8],3)
colors <- densCols(env.dist, act.sore.dist, colramp=clr)
plot(env.dist, act.sore.dist, ylab="Similarity", pch=20,
  xlab="Environmental distance", col = colors, cex=1, cex.lab=1.2)
title(paste("Active: slope = ", slope1), line = 1, cex=1)
abline(DD.act.env, col = "black", lwd=3)


# Make Plot for all pres-abs (env)
slope2 <- round(coefficients(DD.all.env)[2], 3)
int2 <- round(coefficients(DD.all.env)[1], 3)
p <- round(summary(DD.all.env)$coefficients[8], 3)
colors <- densCols(env.dist, all.sore.dist, colramp=clr)
plot(env.dist, all.sore.dist, ylab="Similarity", pch=20,
  xlab="Environmental distance", col = colors, cex=1, cex.lab=1.2)
title(paste("All: slope = ", slope2), line = 1, cex=2)
abline(DD.all.env, col = "black", lwd=3)

# Make Plot for active pres-abs (geo)
slope3 <- round(coefficients(DD.act.geo)[2], 3)
int3 <- round(coefficients(DD.act.geo)[1], 3)
p <- round(summary(DD.act.geo)$coefficients[8],3)
colors <- densCols(geo.dist, act.sore.dist, colramp=clr)
plot(geo.dist, act.sore.dist, ylab="Similarity", pch=20,
  xlab="Geographic distance", col = colors, cex=1, cex.lab=1.2)
title(paste("Active: slope = ", slope3), line = 1, cex=1)
abline(DD.act.geo, col = "black", lwd=3)


# Make Plot for all pres-abs (geo)
slope4 <- round(coefficients(DD.all.geo)[2], 3)
int4 <- round(coefficients(DD.all.geo)[1], 3)
p <- round(summary(DD.all.geo)$coefficients[8], 3)
colors <- densCols(geo.dist, all.sore.dist, colramp=clr)
plot(geo.dist, all.sore.dist, ylab="Similarity", pch=20,
  xlab="Geographic distance", col = colors, cex=1, cex.lab=1.2)
title(paste("All: slope = ", slope4), line = 1, cex=1)
abline(DD.all.geo, col = "black", lwd=3)



# Add X-Axis Label to Plot
a.dd <- 'y'
if(a.dd == 'y'){
  mtext("Geographic Distance", side = 1, adj = 0, outer = TRUE)
  d1 <- diffslope(env.dist, act.sore.dist, env.dist, all.sore.dist, ic=TRUE, permutations=1000)
  d2 <- diffslope(geo.dist, act.sore.dist, geo.dist, all.sore.dist, ic=TRUE, permutations=1000)
  
  Mtitle <- paste("Sorensen\n",
      round(100*(abs((slope1-slope2)/mean(c(slope1, slope2)))),1), '% difference in environmental DDRs; p = ', d1[3],
      '\n',round(100*(abs((slope3-slope4)/mean(c(slope3, slope4)))),1), '% difference in geographic DDRs; p = ', d2[3], sep="")
  title(main=Mtitle, line=0, outer=T, cex.main=1.2)
}
dev.off()

print(c('act.env', 'all.env', 'act.geo', 'all.geo'))
print(c(slope1, slope2, slope3, slope4))
print(c(int1, int2, int3, int4))

100*(abs((int1-int2)/mean(c(int1, int2))))
100*(abs((int3-int4)/mean(c(int3, int4))))
```


```{r, results = 'hide', echo=FALSE, message = FALSE, warning = FALSE, fig.width=6, fig.height=6}

# Regression for active weighted, environmental distance
DD.act.env <- lm(act.canb.dist ~ env.dist)
#summary(DD.act.env, correlation = TRUE)

# Regression for all weighted, environmental distance
DD.all.env <- lm(all.canb.dist ~ env.dist)
coeff <- summary(DD.all.env, correlation = TRUE)

# Regression for active weighted, geographic distance
DD.act.geo <- lm(act.canb.dist ~ geo.dist)
#summary(DD.act.geo, correlation = TRUE)

# Regression for all weighted, geographic distance
DD.all.geo <- lm(all.canb.dist ~ geo.dist)
coeff <- summary(DD.all.geo, correlation = TRUE)

clr <- colorRampPalette(c("#00007F", "blue", "#007FFF", "cyan",
              "#7FFF7F", "yellow", "#FF7F00", "red", "#7F0000"))

file <- paste("~/GitHub/DistDecay/figs/FromSites/DDR-Canberra.png", sep="")
png(filename=file, width=5, height=5, units="in", res=600, pointsize=10)
par(mfrow=c(2, 2), mar = c(5,5,3,1), oma =c(0,0,4,0))

# Make Plot for active pres-abs (env)
slope1 <- round(coefficients(DD.act.env)[2], 3)
int1 <- round(coefficients(DD.act.env)[1], 3)
p <- round(summary(DD.act.env)$coefficients[8],3)
colors <- densCols(env.dist, act.canb.dist, colramp=clr)
plot(env.dist, act.canb.dist, ylab="Similarity", pch=20,
  xlab="Environmental distance", col = colors, cex=1, cex.lab=1.2)
title(paste("Active: slope = ", slope1), line = 1, cex=1)
abline(DD.act.env, col = "black", lwd=3)


# Make Plot for all pres-abs (env)
slope2 <- round(coefficients(DD.all.env)[2], 3)
int2 <- round(coefficients(DD.all.env)[1], 3)
p <- round(summary(DD.all.env)$coefficients[8], 3)
colors <- densCols(env.dist, all.canb.dist, colramp=clr)
plot(env.dist, all.canb.dist, ylab="Similarity", pch=20,
  xlab="Environmental distance", col = colors, cex=1, cex.lab=1.2)
title(paste("All: slope = ", slope2), line = 1, cex=2)
abline(DD.all.env, col = "black", lwd=3)

# Make Plot for active pres-abs (geo)
slope3 <- round(coefficients(DD.act.geo)[2], 3)
int3 <- round(coefficients(DD.act.geo)[1], 3)
p <- round(summary(DD.act.geo)$coefficients[8],3)
colors <- densCols(geo.dist, act.canb.dist, colramp=clr)
plot(geo.dist, act.canb.dist, ylab="Similarity", pch=20,
  xlab="Geographic distance", col = colors, cex=1, cex.lab=1.2)
title(paste("Active: slope = ", slope3), line = 1, cex=1)
abline(DD.act.geo, col = "black", lwd=3)


# Make Plot for all pres-abs (geo)
slope4 <- round(coefficients(DD.all.geo)[2], 3)
int4 <- round(coefficients(DD.all.geo)[1], 3)
p <- round(summary(DD.all.geo)$coefficients[8], 3)
colors <- densCols(geo.dist, all.canb.dist, colramp=clr)
plot(geo.dist, all.canb.dist, ylab="Similarity", pch=20,
  xlab="Geographic distance", col = colors, cex=1, cex.lab=1.2)
title(paste("All: slope = ", slope4), line = 1, cex=1)
abline(DD.all.geo, col = "black", lwd=3)



# Add X-Axis Label to Plot
a.dd <- 'y'
if(a.dd == 'y'){
  mtext("Geographic Distance", side = 1, adj = 0, outer = TRUE)
  d1 <- diffslope(env.dist, act.canb.dist, env.dist, all.canb.dist, ic=TRUE, permutations=1000)
  d2 <- diffslope(geo.dist, act.canb.dist, geo.dist, all.canb.dist, ic=TRUE, permutations=1000)
  
  Mtitle <- paste("Canberra\n",
      round(100*(abs((slope1-slope2)/mean(c(slope1, slope2)))),1), '% difference in environmental DDRs; p = ', d1[3],
      '\n',round(100*(abs((slope3-slope4)/mean(c(slope3, slope4)))),1), '% difference in geographic DDRs; p = ', d2[3], sep="")
  title(main=Mtitle, line=0, outer=T, cex.main=1.2)
}
dev.off()

print(c('act.env', 'all.env', 'act.geo', 'all.geo'))
print(c(slope1, slope2, slope3, slope4))
print(c(int1, int2, int3, int4))

100*(abs((int1-int2)/mean(c(int1, int2))))
100*(abs((int3-int4)/mean(c(int3, int4))))
```








###################################################
# Compare Env & Geo
###################################################


```{r, results = 'hide', echo=FALSE, message = FALSE, warning = FALSE, fig.width=6, fig.height=6}

# Regression for active weighted, environmental distance
DD.act.env <- lm(act.bray.dist ~ env.dist)
#summary(DD.act.env, correlation = TRUE)

# Regression for all weighted, environmental distance
DD.all.env <- lm(all.bray.dist ~ env.dist)
coeff <- summary(DD.all.env, correlation = TRUE)

# Regression for active weighted, geographic distance
DD.act.geo <- lm(act.bray.dist ~ geo.dist)
#summary(DD.act.geo, correlation = TRUE)

# Regression for all weighted, geographic distance
DD.all.geo <- lm(all.bray.dist ~ geo.dist)
coeff <- summary(DD.all.geo, correlation = TRUE)

clr <- colorRampPalette(c("#00007F", "blue", "#007FFF", "cyan",
              "#7FFF7F", "yellow", "#FF7F00", "red", "#7F0000"))

file <- paste("~/GitHub/DistDecay/figs/FromSites/DDR-Bray-EnvVsGeo.png", sep="")
png(filename=file, width=5, height=5, units="in", res=600, pointsize=10)
par(mfrow=c(2, 2), mar = c(5,5,3,1), oma =c(0,0,4,0))

# Make Plot for active pres-abs (env)
slope1 <- round(coefficients(DD.act.env)[2], 3)
int1 <- round(coefficients(DD.act.env)[1], 3)
p <- round(summary(DD.act.env)$coefficients[8],3)
colors <- densCols(env.dist, act.bray.dist, colramp=clr)
plot(env.dist, act.bray.dist, ylab="Similarity", pch=20,
  xlab="Environmental distance", col = colors, cex=1, cex.lab=1.2)
title(paste("Active: slope = ", slope1), line = 1, cex=1)
abline(DD.act.env, col = "black", lwd=3)


# Make Plot for all pres-abs (env)
slope2 <- round(coefficients(DD.all.env)[2], 3)
int2 <- round(coefficients(DD.all.env)[1], 3)
p <- round(summary(DD.all.env)$coefficients[8], 3)
colors <- densCols(env.dist, all.bray.dist, colramp=clr)
plot(env.dist, all.bray.dist, ylab="Similarity", pch=20,
  xlab="Environmental distance", col = colors, cex=1, cex.lab=1.2)
title(paste("All: slope = ", slope2), line = 1, cex=2)
abline(DD.all.env, col = "black", lwd=3)

# Make Plot for active pres-abs (geo)
slope3 <- round(coefficients(DD.act.geo)[2], 3)
int3 <- round(coefficients(DD.act.geo)[1], 3)
p <- round(summary(DD.act.geo)$coefficients[8],3)
colors <- densCols(geo.dist, act.bray.dist, colramp=clr)
plot(geo.dist, act.bray.dist, ylab="Similarity", pch=20,
  xlab="Geographic distance", col = colors, cex=1, cex.lab=1.2)
title(paste("Active: slope = ", slope3), line = 1, cex=1)
abline(DD.act.geo, col = "black", lwd=3)


# Make Plot for all pres-abs (geo)
slope4 <- round(coefficients(DD.all.geo)[2], 3)
int4 <- round(coefficients(DD.all.geo)[1], 3)
p <- round(summary(DD.all.geo)$coefficients[8], 3)
colors <- densCols(geo.dist, all.bray.dist, colramp=clr)
plot(geo.dist, all.bray.dist, ylab="Similarity", pch=20,
  xlab="Geographic distance", col = colors, cex=1, cex.lab=1.2)
title(paste("All: slope = ", slope4), line = 1, cex=1)
abline(DD.all.geo, col = "black", lwd=3)



# Add X-Axis Label to Plot
a.dd <- 'y'
if(a.dd == 'y'){
  mtext("Geographic Distance", side = 1, adj = 0, outer = TRUE)
  d1 <- diffslope(env.dist, act.bray.dist, geo.dist, act.bray.dist, ic=TRUE, permutations=1000)
  d2 <- diffslope(env.dist, all.bray.dist, geo.dist, all.bray.dist, ic=TRUE, permutations=1000)
  
  Mtitle <- paste("Bray-Curtis\n",
      round(100*(abs((slope1-slope3)/mean(c(slope1, slope3)))),1), '% difference in Active DDRs; p = ', d1[3],
      '\n',round(100*(abs((slope2-slope4)/mean(c(slope2, slope4)))),1), '% difference in All DDRs; p = ', d2[3], sep="")
  title(main=Mtitle, line=0, outer=T, cex.main=1.2)
}
dev.off()

print(c('act.env', 'act.geo', 'all.env', 'all.geo'))
print(c(slope1, slope3, slope2, slope4))
print(c(int1, int3, int2, int4))

100*(abs((int1-int3)/mean(c(int1, int3))))
100*(abs((int2-int4)/mean(c(int2, int4))))
```




```{r, results = 'hide', echo=FALSE, message = FALSE, warning = FALSE, fig.width=6, fig.height=6}

# Regression for active weighted, environmental distance
DD.act.env <- lm(act.sore.dist ~ env.dist)
#summary(DD.act.env, correlation = TRUE)

# Regression for all weighted, environmental distance
DD.all.env <- lm(all.sore.dist ~ env.dist)
coeff <- summary(DD.all.env, correlation = TRUE)

# Regression for active weighted, geographic distance
DD.act.geo <- lm(act.sore.dist ~ geo.dist)
#summary(DD.act.geo, correlation = TRUE)

# Regression for all weighted, geographic distance
DD.all.geo <- lm(all.sore.dist ~ geo.dist)
coeff <- summary(DD.all.geo, correlation = TRUE)

clr <- colorRampPalette(c("#00007F", "blue", "#007FFF", "cyan",
              "#7FFF7F", "yellow", "#FF7F00", "red", "#7F0000"))

file <- paste("~/GitHub/DistDecay/figs/FromSites/DDR-Sorensen-EnvVsGeo.png", sep="")
png(filename=file, width=5, height=5, units="in", res=600, pointsize=10)
par(mfrow=c(2, 2), mar = c(5,5,3,1), oma =c(0,0,4,0))

# Make Plot for active pres-abs (env)
slope1 <- round(coefficients(DD.act.env)[2], 3)
int1 <- round(coefficients(DD.act.env)[1], 3)
p <- round(summary(DD.act.env)$coefficients[8],3)
colors <- densCols(env.dist, act.sore.dist, colramp=clr)
plot(env.dist, act.sore.dist, ylab="Similarity", pch=20,
  xlab="Environmental distance", col = colors, cex=1, cex.lab=1.2)
title(paste("Active: slope = ", slope1), line = 1, cex=1)
abline(DD.act.env, col = "black", lwd=3)


# Make Plot for all pres-abs (env)
slope2 <- round(coefficients(DD.all.env)[2], 3)
int2 <- round(coefficients(DD.all.env)[1], 3)
p <- round(summary(DD.all.env)$coefficients[8], 3)
colors <- densCols(env.dist, all.sore.dist, colramp=clr)
plot(env.dist, all.sore.dist, ylab="Similarity", pch=20,
  xlab="Environmental distance", col = colors, cex=1, cex.lab=1.2)
title(paste("All: slope = ", slope2), line = 1, cex=2)
abline(DD.all.env, col = "black", lwd=3)

# Make Plot for active pres-abs (geo)
slope3 <- round(coefficients(DD.act.geo)[2], 3)
int3 <- round(coefficients(DD.act.geo)[1], 3)
p <- round(summary(DD.act.geo)$coefficients[8],3)
colors <- densCols(geo.dist, act.sore.dist, colramp=clr)
plot(geo.dist, act.sore.dist, ylab="Similarity", pch=20,
  xlab="Geographic distance", col = colors, cex=1, cex.lab=1.2)
title(paste("Active: slope = ", slope3), line = 1, cex=1)
abline(DD.act.geo, col = "black", lwd=3)


# Make Plot for all pres-abs (geo)
slope4 <- round(coefficients(DD.all.geo)[2], 3)
int4 <- round(coefficients(DD.all.geo)[1], 3)
p <- round(summary(DD.all.geo)$coefficients[8], 3)
colors <- densCols(geo.dist, all.sore.dist, colramp=clr)
plot(geo.dist, all.sore.dist, ylab="Similarity", pch=20,
  xlab="Geographic distance", col = colors, cex=1, cex.lab=1.2)
title(paste("All: slope = ", slope4), line = 1, cex=1)
abline(DD.all.geo, col = "black", lwd=3)



# Add X-Axis Label to Plot
a.dd <- 'y'
if(a.dd == 'y'){
  mtext("Geographic Distance", side = 1, adj = 0, outer = TRUE)
  d1 <- diffslope(env.dist, act.sore.dist, geo.dist, act.sore.dist, ic=TRUE, permutations=1000)
  d2 <- diffslope(env.dist, all.sore.dist, geo.dist, all.sore.dist, ic=TRUE, permutations=1000)
  
  Mtitle <- paste("Sorensen\n",
      round(100*(abs((slope1-slope3)/mean(c(slope1, slope3)))),1), '% difference in Active DDRs; p = ', d1[3],
      '\n',round(100*(abs((slope2-slope4)/mean(c(slope2, slope4)))),1), '% difference in All DDRs; p = ', d2[3], sep="")
  title(main=Mtitle, line=0, outer=T, cex.main=1.2)
}
dev.off()

print(c('act.env', 'act.geo', 'all.env', 'all.geo'))
print(c(slope1, slope3, slope2, slope4))
print(c(int1, int3, int2, int4))

100*(abs((int1-int3)/mean(c(int1, int3))))
100*(abs((int2-int4)/mean(c(int2, int4))))
```


```{r, results = 'hide', echo=FALSE, message = FALSE, warning = FALSE, fig.width=6, fig.height=6}

# Regression for active weighted, environmental distance
DD.act.env <- lm(act.canb.dist ~ env.dist)
#summary(DD.act.env, correlation = TRUE)

# Regression for all weighted, environmental distance
DD.all.env <- lm(all.canb.dist ~ env.dist)
coeff <- summary(DD.all.env, correlation = TRUE)

# Regression for active weighted, geographic distance
DD.act.geo <- lm(act.canb.dist ~ geo.dist)
#summary(DD.act.geo, correlation = TRUE)

# Regression for all weighted, geographic distance
DD.all.geo <- lm(all.canb.dist ~ geo.dist)
coeff <- summary(DD.all.geo, correlation = TRUE)

clr <- colorRampPalette(c("#00007F", "blue", "#007FFF", "cyan",
              "#7FFF7F", "yellow", "#FF7F00", "red", "#7F0000"))

file <- paste("~/GitHub/DistDecay/figs/FromSites/DDR-Canberra-EnvVsGeo.png", sep="")
png(filename=file, width=5, height=5, units="in", res=600, pointsize=10)
par(mfrow=c(2, 2), mar = c(5,5,3,1), oma =c(0,0,4,0))

# Make Plot for active pres-abs (env)
slope1 <- round(coefficients(DD.act.env)[2], 3)
int1 <- round(coefficients(DD.act.env)[1], 3)
p <- round(summary(DD.act.env)$coefficients[8],3)
colors <- densCols(env.dist, act.canb.dist, colramp=clr)
plot(env.dist, act.canb.dist, ylab="Similarity", pch=20,
  xlab="Environmental distance", col = colors, cex=1, cex.lab=1.2)
title(paste("Active: slope = ", slope1), line = 1, cex=1)
abline(DD.act.env, col = "black", lwd=3)


# Make Plot for all pres-abs (env)
slope2 <- round(coefficients(DD.all.env)[2], 3)
int2 <- round(coefficients(DD.all.env)[1], 3)
p <- round(summary(DD.all.env)$coefficients[8], 3)
colors <- densCols(env.dist, all.canb.dist, colramp=clr)
plot(env.dist, all.canb.dist, ylab="Similarity", pch=20,
  xlab="Environmental distance", col = colors, cex=1, cex.lab=1.2)
title(paste("All: slope = ", slope2), line = 1, cex=2)
abline(DD.all.env, col = "black", lwd=3)

# Make Plot for active pres-abs (geo)
slope3 <- round(coefficients(DD.act.geo)[2], 3)
int3 <- round(coefficients(DD.act.geo)[1], 3)
p <- round(summary(DD.act.geo)$coefficients[8],3)
colors <- densCols(geo.dist, act.canb.dist, colramp=clr)
plot(geo.dist, act.canb.dist, ylab="Similarity", pch=20,
  xlab="Geographic distance", col = colors, cex=1, cex.lab=1.2)
title(paste("Active: slope = ", slope3), line = 1, cex=1)
abline(DD.act.geo, col = "black", lwd=3)


# Make Plot for all pres-abs (geo)
slope4 <- round(coefficients(DD.all.geo)[2], 3)
int4 <- round(coefficients(DD.all.geo)[1], 3)
p <- round(summary(DD.all.geo)$coefficients[8], 3)
colors <- densCols(geo.dist, all.canb.dist, colramp=clr)
plot(geo.dist, all.canb.dist, ylab="Similarity", pch=20,
  xlab="Geographic distance", col = colors, cex=1, cex.lab=1.2)
title(paste("All: slope = ", slope4), line = 1, cex=1)
abline(DD.all.geo, col = "black", lwd=3)



# Add X-Axis Label to Plot
a.dd <- 'y'
if(a.dd == 'y'){
  mtext("Geographic Distance", side = 1, adj = 0, outer = TRUE)
  d1 <- diffslope(env.dist, act.canb.dist, geo.dist, act.canb.dist, ic=TRUE, permutations=1000)
  d2 <- diffslope(env.dist, all.canb.dist, geo.dist, all.canb.dist, ic=TRUE, permutations=1000)
  
  Mtitle <- paste("Canberra\n",
      round(100*(abs((slope1-slope3)/mean(c(slope1, slope3)))),1), '% difference in Active DDRs; p = ', d1[3],
      '\n',round(100*(abs((slope2-slope4)/mean(c(slope2, slope4)))),1), '% difference in All DDRs; p = ', d2[3], sep="")
  title(main=Mtitle, line=0, outer=T, cex.main=1.2)
}
dev.off()

print(c('act.env', 'act.geo', 'all.env', 'all.geo'))
print(c(slope1, slope3, slope2, slope4))
print(c(int1, int3, int2, int4))

100*(abs((int1-int3)/mean(c(int1, int3))))
100*(abs((int2-int4)/mean(c(int2, int4))))
```
