---
title: "Supplemental Figures: Compositional similarity vs. Geographic distance"
geometry: margin=2.54cm
date: "November 7, 2015"
output: pdf_document
header-includes: \usepackage{array}
---

```{r, results = 'hide', echo=FALSE, message = FALSE, warning = FALSE}
# Retrieve and set the working directory
rm(list=ls())
getwd()
setwd("~/GitHub/DistDecay/analyses/DistanceDecay")

# Load packages
require("simba")
require("fossil")
require("dplyr")
load(file = "~/GitHub/DistDecay/data/MicrobialCommunity/INPond_Initial.RData")
#load(file = "~/GitHub/DistDecay/data/DistDecayDom-1x2-Data.RData")

# Load Environmental and Geographical Data
env <- read.table("~/GitHub/DistDecay/data/Environmental/20130801_PondDataMod.csv", sep = ",", header = TRUE)
env <- env[complete.cases(env),]
env[which(env$Sample_ID == "HNF133"), ]["SpC"] <- 55
#env$chla <- log(env$chla)
#env$TP <- log(env$TP)
#env$canopy <- log(env$canopy)
#env$DON <- log(env$DON)
#env$DOC <- log(env$DOC)
#env$Color <- log(env$Color)
env[which(env$Sample_ID == "YSF46"), ]["lat"] <- 39.1186
```


```{r, results = 'hide', echo=FALSE, message = FALSE, warning = FALSE}
# Select DNA Data: Use the `grep()` Command and Rename with `gsub()`
# The active portion, based on cDNA
act.com <- Pond97[grep("*-cDNA", rownames(Pond97)), ]
rownames(act.com) <- gsub("\\-cDNA", "", rownames(act.com))

# The community without respect to active or not, 16S rRNA gene sequences
all.com <- Pond97[grep("*-DNA", rownames(Pond97)), ]
rownames(all.com) <- gsub("\\-DNA", "", rownames(all.com))

# Remove Sites Not in the Environmental Data Set
act.com <- act.com[rownames(act.com)  %in% env$Sample_ID, ]
all.com <- all.com[rownames(all.com)  %in% env$Sample_ID, ]

# Remove sites not shared between matrices
all.com <- all.com[rownames(all.com)  %in% row.names(act.com),]
env <- env[env$Sample_ID %in% row.names(act.com),]

# Remove Zero-Occurrence Taxa
act.com2 <- act.com[,colSums(act.com) > 0]
all.com2 <- all.com[,colSums(all.com) > 0]

# Use relative abundances
#act.com <- sweep(act.com, 1, rowSums(act.com), '/')
#all.com <- sweep(all.com, 1, rowSums(all.com), '/')

#act.com2 <- sweep(act.com2, 1, rowSums(act.com2), '/')
#all.com2 <- sweep(all.com2, 1, rowSums(all.com2), '/')
```


```{r, results = 'hide', echo=FALSE, message = FALSE, warning = FALSE, fig.width=6, fig.height=6}
# Use n most abundant taxa
metric <- "bray"
ns <- c(10, 32, 100, 316, 1000, 3162, 10000, dim(act.com)[2])

df.all.env.bray.1 <- as.data.frame(setNames(replicate(length(ns), numeric(0)), sprintf("%.f", ns)))
df.all.geo.bray.1 <- as.data.frame(setNames(replicate(length(ns), numeric(0)), sprintf("%.f", ns)))
df.act.env.bray.1 <- as.data.frame(setNames(replicate(length(ns), numeric(0)), sprintf("%.f", ns)))
df.act.geo.bray.1 <- as.data.frame(setNames(replicate(length(ns), numeric(0)), sprintf("%.f", ns)))

for (i in c(1:1)){
  print(i)
  all.env.bray <- c() 
  all.geo.bray <- c()
  act.env.bray <- c()
  act.geo.bray <- c()

  for (n in ns){
    #rows <- sample(c(1:49), 49, replace=FALSE)
    env.dat <- env
    
    # Geographic Distances (Kilometers) Among Ponds
    long.lat <- as.matrix(cbind(env.dat$long, env.dat$lat))
    geo.dist <- earth.dist(long.lat, dist = TRUE)
    geo.dist <- geo.dist/max(geo.dist)
    geo.dist[which(!is.finite(geo.dist))] = NA

    env.dat <- as.matrix(subset(env, select = c("canopy", "DO", "ORP", "pH", "Temp",
              "DOC", "DON", "TDS", "Color", "chla", "TP", "Depth", "Salinity", "Diameter")))
    env.dat <- as.matrix(env.dat)
    env.dat <- scale(env.dat)
    
    # Conduct PCA
    pca <- princomp(env.dat)
    scores <- as.data.frame(pca$scores)
    env.dist <- vegdist(scores[, 1], "euclidean")
    env.dist <- env.dist/max(env.dist)
    env.dist[which(!is.finite(env.dist))] = NA

    all.com2 <- all.com[, 1:n]
    act.com2 <- act.com[, 1:n]
  
    #act.com2 <- act.com2[,order(colMeans(-act.com2))]
    #act.com2 <- act.com2[,order(colSums(-act.com2))]
    #act.com2 <- act.com2[,order(colSums(-act.com2==0))]

    #all.com2 <- all.com2[,order(colMeans(-all.com2))]
    #all.com2 <- all.com2[,order(colSums(-all.com2))]
    #all.com2 <- all.com2[,order(colSums(-all.com2==0))]

    # Use relative abundance
    act.com2 <- sweep(act.com2, 1, rowSums(act.com2), '/')
    all.com2 <- sweep(all.com2, 1, rowSums(all.com2), '/')
  
    # Taxonomic Distances Among Ponds
    act.weighted.dist <- 1 - vegdist(act.com2, method=metric, binary=F)
    all.weighted.dist <- 1 - vegdist(all.com2, method=metric, binary=F)
  
    # Regression for active weighted
    DD.act.env <- lm(act.weighted.dist ~ env.dist)
    act.env.bray <- c(act.env.bray, round(coefficients(DD.act.env)[2], 4))
  
    # Regression for all weighted
    DD.all.env <- lm(all.weighted.dist ~ env.dist)
    all.env.bray <- c(all.env.bray, round(coefficients(DD.all.env)[2], 4))
  
    # Regression for active weighted
    DD.act.geo <- lm(act.weighted.dist ~ geo.dist)
    act.geo.bray <- c(act.geo.bray, round(coefficients(DD.act.geo)[2], 4))
  
    # Regression for all weighted
    DD.all.geo <- lm(all.weighted.dist ~ geo.dist)
    all.geo.bray <- c(all.geo.bray, round(coefficients(DD.all.geo)[2], 4))
  
  }
    
  df.all.env.bray.1 <- rbind(df.all.env.bray.1, all.env.bray)
  df.all.geo.bray.1 <- rbind(df.all.geo.bray.1, all.geo.bray)
  df.act.env.bray.1 <- rbind(df.act.env.bray.1, act.env.bray)
  df.act.geo.bray.1 <- rbind(df.act.geo.bray.1, act.geo.bray)
}
```



```{r, results = 'hide', echo=FALSE, message = FALSE, warning = FALSE, fig.width=6, fig.height=6}
# Use n most abundant taxa
metric <- "bray"
ns <- c(13, 18, 23, 28, 33, 38, 43, 48)

df.all.env.bray.2 <- as.data.frame(setNames(replicate(length(ns), numeric(0)), sprintf("%.f", ns)))
df.all.geo.bray.2 <- as.data.frame(setNames(replicate(length(ns), numeric(0)), sprintf("%.f", ns)))
df.act.env.bray.2 <- as.data.frame(setNames(replicate(length(ns), numeric(0)), sprintf("%.f", ns)))
df.act.geo.bray.2 <- as.data.frame(setNames(replicate(length(ns), numeric(0)), sprintf("%.f", ns)))

for (i in c(1:10)){
  print(i)
  all.env.bray <- c() 
  all.geo.bray <- c()
  act.env.bray <- c()
  act.geo.bray <- c()

  for (n in ns){
    rows <- as.numeric(sample(c(1:49), n, replace=FALSE))
    env.dat <- env[rows, ]
    
    # Geographic Distances (Kilometers) Among Ponds
    long.lat <- as.matrix(cbind(env.dat$long, env.dat$lat))
    geo.dist <- earth.dist(long.lat, dist = TRUE)
    geo.dist <- geo.dist/max(geo.dist)
    geo.dist[which(!is.finite(geo.dist))] = NA

    env.dat <- as.matrix(subset(env.dat, select = c("canopy", "DO", "ORP", "pH",
      "Temp", "DOC", "DON", "TDS", "Color", "chla", "TP", "Depth", "Salinity")))
    env.dat <- as.matrix(env.dat)
    env.dat <- scale(env.dat)
    
    # Conduct PCA
    pca <- princomp(env.dat)
    scores <- as.data.frame(pca$scores)
    env.dist <- vegdist(scores[, 1], "euclidean")
    env.dist <- env.dist/max(env.dist)
    env.dist[which(!is.finite(env.dist))] = NA

    all.com2 <- all.com[rows, ]
    act.com2 <- act.com[rows, ]
  
    # Use relative abundance
    act.com2 <- sweep(act.com2, 1, rowSums(act.com2), '/')
    all.com2 <- sweep(all.com2, 1, rowSums(all.com2), '/')
  
    # Taxonomic Distances Among Ponds
    act.weighted.dist <- 1 - vegdist(act.com2, method=metric, binary=F)
    all.weighted.dist <- 1 - vegdist(all.com2, method=metric, binary=F)
  
    # Regression for active weighted
    DD.act.env <- lm(act.weighted.dist ~ env.dist)
    act.env.bray <- c(act.env.bray, round(coefficients(DD.act.env)[2], 4))
  
    # Regression for all weighted
    DD.all.env <- lm(all.weighted.dist ~ env.dist)
    all.env.bray <- c(all.env.bray, round(coefficients(DD.all.env)[2], 4))
  
    # Regression for active weighted
    DD.act.geo <- lm(act.weighted.dist ~ geo.dist)
    act.geo.bray <- c(act.geo.bray, round(coefficients(DD.act.geo)[2], 4))
  
    # Regression for all weighted
    DD.all.geo <- lm(all.weighted.dist ~ geo.dist)
    all.geo.bray <- c(all.geo.bray, round(coefficients(DD.all.geo)[2], 4))
  }
  
  df.all.env.bray.2 <- rbind(df.all.env.bray.2, all.env.bray)
  df.all.geo.bray.2 <- rbind(df.all.geo.bray.2, all.geo.bray)
  df.act.env.bray.2 <- rbind(df.act.env.bray.2, act.env.bray)
  df.act.geo.bray.2 <- rbind(df.act.geo.bray.2, act.geo.bray)
}
```


```{r, results = 'hide', echo=FALSE, message = FALSE, warning = FALSE, fig.width=6, fig.height=6}
plot.new()
file <- paste("~/GitHub/DistDecay/figs/Fig2-DDR-v2x2-SitesAndOTUs.png", sep="")
png(filename=file, width=6, height=6, units="in", res=600, pointsize=10)

par(mfrow=c(2, 2), mar = c(5,5,3,1), oma =c(0,0,0,0))

xs <- log10(c(10, 32, 100, 316, 1000, 3162, 10000, dim(act.com)[2]))
xlabel <- "Number of OTUs, log10"

# Make Plot for Environmental DDRs
avg <- colMeans(df.all.env.bray.1)
sem <- sapply(df.all.env.bray.1, function(x)sd(x)/sqrt(length(x)))
plot(xs, colMeans(df.all.env.bray.1), ylab="DDR slope", xlab=xlabel,
  col = 'blue', cex=1, cex.lab=1.2, cex.axis = 1, pch=16, ylim=c(-0.6, -0.2))
text(2.2, -0.25, labels = 'total', cex = 1, col = 'blue')
arrows(xs, avg - sem, xs, avg + sem, length = 0.05, angle=90, code=3, col='blue')

avg <- colMeans(df.act.env.bray.1)
sem <- sapply(df.act.env.bray.1, function(x)sd(x)/sqrt(length(x)))
points(xs, colMeans(df.act.env.bray.1), ylab="DDR slope", xlab=xlabel,
  col = 'red', cex=1, cex.lab=1.2, cex.axis = 1, pch=16)
text(2.2, -0.42, labels = 'active', cex = 1, col = 'red')
arrows(xs, avg - sem, xs, avg + sem, length = 0.05, angle=90, code=3, col='red')
title(paste("Environmental"), line = 1, cex=2)


# Make Plot for Geographic DDRs
avg <- colMeans(df.all.geo.bray.1)
sem <- sapply(df.all.geo.bray.1, function(x)sd(x)/sqrt(length(x)))
plot(xs, colMeans(df.all.geo.bray.1), ylab="DDR slope", xlab=xlabel,
  col = 'blue', cex=1, cex.lab=1.2, cex.axis = 1, pch=16,  ylim=c(-0.35, -0.1))
text(2.2, -0.125, labels = 'total', cex = 1, col = 'blue')
arrows(xs, avg - sem, xs, avg + sem, length = 0.05, angle=90, code=3, col='blue')

avg <- colMeans(df.act.geo.bray.1)
sem <- sapply(df.act.geo.bray.1, function(x)sd(x)/sqrt(length(x)))
points(xs, colMeans(df.act.geo.bray.1), ylab="DDR slope", xlab=xlabel,
  col = 'red', cex=1, cex.lab=1.2, cex.axis = 1, pch=16)
text(2.2, -0.22, labels = 'active', cex = 1, col = 'red')
arrows(xs, avg - sem, xs, avg + sem, length = 0.05, angle=90, code=3, col='red')
title(paste("Geographic"), line = 1, cex=2)


xs <- c(13, 18, 23, 28, 33, 38, 43, 48)
xlabel <- "Number of sites"

# Make Plot for Environmental DDRs
avg <- colMeans(df.all.env.bray.2)
sem <- sapply(df.all.env.bray.2, function(x)sd(x)/sqrt(length(x)))
plot(xs, colMeans(df.all.env.bray.2), ylab="DDR slope", xlab=xlabel,
  col = 'blue', cex=1, cex.lab=1.2, cex.axis = 1, pch=16, ylim=c(-0.35, -0.1))
#text(2.2, -0.38, labels = 'total', cex = 1, col = 'blue')
arrows(xs, avg - sem, xs, avg + sem, length = 0.05, angle=90, code=3, col='blue')

avg <- colMeans(df.act.env.bray.2)
sem <- sapply(df.act.env.bray.2, function(x)sd(x)/sqrt(length(x)))
points(xs, colMeans(df.act.env.bray.2), ylab="DDR slope", xlab=xlabel,
  col = 'red', cex=1, cex.lab=1.2, cex.axis = 1, pch=16)
#text(2.2, -0.6, labels = 'active', cex = 1, col = 'red')
arrows(xs, avg - sem, xs, avg + sem, length = 0.05, angle=90, code=3, col='red')
title(paste("Environmental"), line = 1, cex=2)


# Make Plot for Geographic DDRs
avg <- colMeans(df.all.geo.bray.2)
sem <- sapply(df.all.geo.bray.2, function(x)sd(x)/sqrt(length(x)))
plot(xs, colMeans(df.all.geo.bray.2), ylab="DDR slope", xlab=xlabel,
  col = 'blue', cex=1, cex.lab=1.2, cex.axis = 1, pch=16,  ylim=c(-0.22, -0.1))
#text(2.2, -0.085, labels = 'total', cex = 1, col = 'blue')
arrows(xs, avg - sem, xs, avg + sem, length = 0.05, angle=90, code=3, col='blue')

avg <- colMeans(df.act.geo.bray.2)
sem <- sapply(df.act.geo.bray.2, function(x)sd(x)/sqrt(length(x)))
points(xs, colMeans(df.act.geo.bray.2), ylab="DDR slope", xlab=xlabel,
  col = 'red', cex=1, cex.lab=1.2, cex.axis = 1, pch=16)
#text(2.2, -0.115, labels = 'active', cex = 1, col = 'red')
arrows(xs, avg - sem, xs, avg + sem, length = 0.05, angle=90, code=3, col='red')
title(paste("Geographic"), line = 1, cex=2)

dev.off()
```