---
title: "Distance Decay Relationship among microbial communities of IN Ponds"
geometry: margin=2.54cm
date: "November 7, 2015"
output: pdf_document
header-includes: \usepackage{array}
---

```{r}
# Retrieve and set the working directory
rm(list=ls())
getwd()
setwd("~/GitHub/DistDecay/analyses/Beta-Div/DistanceDecay")

# Load packages
require("simba")
require("fossil")
load(file = "~/GitHub/DistDecay/data/MicrobialCommunity/INPond_Initial.RData")

# Load Environmental and Geographical Data
env <- read.table("~/GitHub/DistDecay/data/Environmental/20130801_PondDataMod.csv", 
    sep = ",", header = TRUE)
env <- env[complete.cases(env),]
env[which(env$Sample_ID == "HNF133"), ]["SpC"] <- 55
env$chla <- log(env$chla)
```


```{r, results = 'hide', echo=FALSE, message = FALSE, warning = FALSE}
# Select DNA Data: Use the `grep()` Command and Rename with `gsub()`
# The active portion, based on cDNA
act.com <- Pond97[grep("*-cDNA", rownames(Pond97)), ]
rownames(act.com) <- gsub("\\-cDNA", "", rownames(act.com))

# The community without respect to active or not, 16S rRNA gene sequences
all.com <- Pond97[grep("*-DNA", rownames(Pond97)), ]
rownames(all.com) <- gsub("\\-DNA", "", rownames(all.com))

# Remove Sites Not in the Environmental Data Set
act.com <- act.com[rownames(act.com)  %in% env$Sample_ID, ]
all.com <- all.com[rownames(all.com)  %in% env$Sample_ID, ]

# Remove sites not shared between matrices
all.com <- all.com[rownames(all.com)  %in% row.names(act.com),]
env <- env[env$Sample_ID %in% row.names(act.com),]

# Remove Zero-Occurrence Taxa
act.com <- act.com[ , colSums(act.com) > 0]
all.com <- all.com[ , colSums(all.com) > 0]

# Use relative abundance
act.com <- sweep(act.com, 1, rowSums(act.com), '/')
all.com <- sweep(all.com, 1, rowSums(all.com), '/')
```



```{r}
# Geographic Distances (Kilometers) Among Ponds
long.lat <- as.matrix(cbind(env$long, env$lat))
geo.dist <- earth.dist(long.lat, dist = TRUE)
geo.dist <- log(geo.dist)
geo.dist[which(!is.finite(geo.dist))] = NA

# Define Environmental Matrix
env.dat <- as.matrix(subset(env, select = c("canopy", "DO", "ORP", "pH", "Temp", "DOC", "DON", "TDS", "Color", "chla", "TP", "Depth", "Salinity")))
#env.dat <- as.matrix(subset(env, select = c("DOC", "Temp", "chla", "Depth", "pH", "DON")))
env.dat <- scale(env.dat)

# Conduct PCA
pca <- princomp(env.dat)
scores <- as.data.frame(pca$scores)
env.dist <- log(vegdist(scores, "euclidean"))
env.dist[which(!is.finite(env.dist))] = NA
```


```{r}
metric <- "bray"
# Taxonomic Distances Among Ponds
act.weighted.dist <- log(1 - vegdist(act.com, method=metric, binary=F))
all.weighted.dist <- log(1 - vegdist(all.com, method=metric, binary=F))

act.presabs.dist <- log(1 - vegdist(act.com, method=metric, binary=T))
all.presabs.dist <- log(1 - vegdist(all.com, method=metric, binary=T))
```


```{r}
# Regression for active weighted, environmental distance
DD.act.env <- lm(act.weighted.dist ~ env.dist)
summary(DD.act.env, correlation = TRUE)

# Regression for all weighted, environmental distance
DD.all.env <- lm(all.weighted.dist ~ env.dist)
coeff <- summary(DD.all.env, correlation = TRUE)

# Regression for active weighted, geographic distance
DD.act.geo <- lm(act.weighted.dist ~ geo.dist)
summary(DD.act.geo, correlation = TRUE)

# Regression for all weighted, geographic distance
DD.all.geo <- lm(all.weighted.dist ~ geo.dist)
coeff <- summary(DD.all.geo, correlation = TRUE)
```


```{r, results = 'hide', echo=FALSE, message = FALSE, warning = FALSE, fig.width=6, fig.height=6}
file <- paste("~/GitHub/DistDecay/figs/DDR-Bray.png", sep="")
png(filename=file, width=5, height=5, units="in", res=600, pointsize=10)
par(mfrow=c(2, 2), mar = c(5,5,3,1), oma =c(0,0,6,0))

# Make Plot for all pres-abs (env)
slope1 <- round(coefficients(DD.all.env)[2], 3)
p <- round(summary(DD.all.env)$coefficients[8], 3)
smoothScatter(env.dist, all.weighted.dist, ylab="log(Similarity)",
  xlab="log(Env. distance)", col = NULL, cex=1, cex.lab=1.2, 
  cex.axis = 1, colramp = colorRampPalette(c("white", "white", 
  "#007FFF", "cyan", "#7FFF7F", "yellow", "#FF7F00", "red", "#7F0000")))
title(paste("All: slope = ", slope1), line = 1, cex=2)
abline(DD.all.env, col = "black", lwd=3)

# Make Plot for active pres-abs (env)
slope2 <- round(coefficients(DD.act.env)[2], 3)
p <- round(summary(DD.act.env)$coefficients[8],3)
smoothScatter(env.dist, act.weighted.dist, 
    ylab="log(Similarity)", xlab="log(Env. distance)",
  col = NULL, cex=1, cex.lab=1.2, cex.axis = 1,
  colramp = colorRampPalette(c("white", "white", "#007FFF", 
  "cyan", "#7FFF7F", "yellow", "#FF7F00", "red", "#7F0000")))
title(paste("Active: slope = ", slope2), line = 1, cex=1)
abline(DD.act.env, col = "black", lwd=3)

# Make Plot for all pres-abs (geo)
slope3 <- round(coefficients(DD.all.geo)[2], 3)
p <- round(summary(DD.all.geo)$coefficients[8], 3)
smoothScatter(geo.dist, all.weighted.dist, 
  ylab="log(Similarity)", xlab="log(Distance), km",
  col = NULL, cex=1, cex.lab=1.2, cex.axis = 1,
  colramp = colorRampPalette(c("white", "white", "#007FFF",
  "cyan", "#7FFF7F", "yellow", "#FF7F00", "red", "#7F0000")))
title(paste("All: slope = ", slope3), line = 1, cex=1)
abline(DD.all.geo, col = "black", lwd=3)

# Make Plot for active pres-abs (geo)
slope4 <- round(coefficients(DD.act.geo)[2], 3)
p <- round(summary(DD.act.geo)$coefficients[8],3)
smoothScatter(geo.dist, act.weighted.dist, 
  ylab="log(Similarity)", xlab="log(Distance), km",
  col = NULL, cex=1, cex.lab=1.2, cex.axis = 1,
  colramp = colorRampPalette(c("white", "white", "#007FFF",
  "cyan", "#7FFF7F", "yellow", "#FF7F00", "red", "#7F0000")))
title(paste("Active: slope = ", slope4), line = 1, cex=1)
abline(DD.act.geo, col = "black", lwd=3)

# Add X-Axis Label to Plot
mtext("Geographic Distance, km", side = 1, adj = 0, outer = TRUE)
d1 <- diffslope(env.dist, act.weighted.dist, env.dist, all.weighted.dist, ic=TRUE, permutations=1000)
d2 <- diffslope(geo.dist, act.weighted.dist, geo.dist, all.weighted.dist, ic=TRUE, permutations=1000)

Mtitle <- paste("Abundance weighted similarity (Bray-Curtis)\n",
    round(100*(abs((slope1-slope2)/mean(c(slope1, slope2)))),1), '% difference in environmental DDRs; p = ', d1[3],
    '\n',round(100*(abs((slope3-slope4)/mean(c(slope3, slope4)))),1), '% difference in geographic DDRs; p = ', d2[3], sep="")
title(main=Mtitle, line=1, outer=T, cex.main=1.2)
dev.off()
```


```{r, results = 'hide', echo=FALSE, message = FALSE, warning = FALSE}
# Regression for active presabs
DD.act.env <- lm(act.presabs.dist ~ env.dist)
summary(DD.act.env, correlation = TRUE)

# Regression for all presabs
DD.all.env <- lm(all.presabs.dist ~ env.dist)
coeff <- summary(DD.all.env, correlation = TRUE)

# Regression for active presabs
DD.act.geo <- lm(act.presabs.dist ~ geo.dist)
summary(DD.act.geo, correlation = TRUE)

# Regression for all presabs
DD.all.geo <- lm(all.presabs.dist ~ geo.dist)
coeff <- summary(DD.all.geo, correlation = TRUE)
```


```{r, results = 'hide', echo=FALSE, message = FALSE, warning = FALSE, fig.width=6, fig.height=6}
file <- paste("~/GitHub/DistDecay/figs/DDR-Sorensen.png", sep="")
png(filename=file, width=5, height=5, units="in", res=600, pointsize=10)
par(mfrow=c(2, 2), mar = c(5,5,3,1), oma =c(0,0,6,0))

# Make Plot for all pres-abs (env)
slope1 <- round(coefficients(DD.all.env)[2], 3)
p <- round(summary(DD.all.env)$coefficients[8], 3)
smoothScatter(env.dist, all.presabs.dist, ylab="log(Similarity)", xlab="log(Env. distance)",
  col = NULL, cex=1, cex.lab=1.2, cex.axis = 1,
  colramp = colorRampPalette(c("white", "white", "#007FFF", "cyan",
                     "#7FFF7F", "yellow", "#FF7F00", "red", "#7F0000")))
title(paste("All: slope = ", slope1), line = 1, cex=1)
abline(DD.all.env, col = "black", lwd=3)

# Make Plot for active pres-abs (env)
slope2 <- round(coefficients(DD.act.env)[2], 3)
p <- round(summary(DD.act.env)$coefficients[8],3)
smoothScatter(env.dist, act.presabs.dist, ylab="log(Similarity)", xlab="log(Env. distance)",
  col = NULL, cex=1, cex.lab=1.2, cex.axis = 1,
  colramp = colorRampPalette(c("white", "white", "#007FFF", "cyan",
                     "#7FFF7F", "yellow", "#FF7F00", "red", "#7F0000")))
title(paste("Active: slope = ", slope2), line = 1, cex=1)
abline(DD.act.env, col = "black", lwd=3)

# Make Plot for all pres-abs (geo)
slope3 <- round(coefficients(DD.all.geo)[2], 3)
p <- round(summary(DD.all.geo)$coefficients[8], 3)
smoothScatter(geo.dist, all.presabs.dist, ylab="log(Similarity)", xlab="log(Distance), km",
  col = NULL, cex=1, cex.lab=1.2, cex.axis = 1,
  colramp = colorRampPalette(c("white", "white", "#007FFF", "cyan",
                     "#7FFF7F", "yellow", "#FF7F00", "red", "#7F0000")))
title(paste("All: slope = ", slope3), line = 1, cex=1)
abline(DD.all.geo, col = "black", lwd=3)

# Make Plot for active pres-abs (geo)
slope4 <- round(coefficients(DD.act.geo)[2], 3)
p <- round(summary(DD.act.geo)$coefficients[8],3)
smoothScatter(geo.dist, act.presabs.dist, ylab="log(Similarity)", xlab="log(Distance), km",
  col = NULL, cex=1, cex.lab=1.2, cex.axis = 1,
  colramp = colorRampPalette(c("white", "white", "#007FFF", "cyan",
                     "#7FFF7F", "yellow", "#FF7F00", "red", "#7F0000")))
title(paste("Active: slope = ", slope4), line = 1, cex=1)
abline(DD.act.geo, col = "black", lwd=3)

# Add X-Axis Label to Plot
mtext("Geographic Distance, km", side = 1, adj = 0, outer = TRUE)
d1 <- diffslope(env.dist, act.presabs.dist, env.dist, all.presabs.dist, ic=TRUE, permutations=1000)
d2 <- diffslope(geo.dist, act.presabs.dist, geo.dist, all.presabs.dist, ic=TRUE, permutations=1000)

Mtitle <- paste("Similarity based on presence (Sorensen)\n",
    round(100*(abs((slope1-slope2)/mean(c(slope1, slope2)))),1), '% difference in environmental DDRs; p = ', d1[3],
    '\n',round(100*(abs((slope3-slope4)/mean(c(slope3, slope4)))),1), '% difference in geographic DDRs; p = ', d2[3], sep="")
title(main=Mtitle, line=1, outer=T, cex.main=1.2)
dev.off()
```
